### ISSUE TITLE
[Bug Report] Conv2d accuracy issues on resnet50 conv (wormhole), caused by split reader

### ISSUE THREAD

**Issue by pavlepopovic** [tenstorrent/tt-metal#17662]
**Describe the bug**
Follow up from https://github.com/tenstorrent/tt-metal/issues/16895
If split reader is turned on, and act_block_h_override is 49/7/14 * 32, in resnet50 demo on WH, we have an accuracy drop. If act_block_h_override is 1 * 32, we have a crash. 2 * 32 is the only one that seems to work

**To Reproduce**

Attaching a unit test with values that don't work and one that does work. (You can also uncomment the value in models/demos/ttnn_resnet/tt/ttnn_functional_resnet50.py file, and run dataset validation demo, accuracy should drop to ~0.36)

```
import pytest
import torch
import ttnn

from loguru import logger

from tests.ttnn.utils_for_testing import check_with_pcc_without_tensor_printout


@pytest.mark.parametrize(
    "batch, groups, input_channels, output_channels, input_height, input_width, weights_dtype, activations_dtype, kernel, stride, padding, input_channels_alignment, act_block_h_override, act_block_w_div, deallocate_activation, output_layout, math_fidelity, fp32_accum, packer_l1_acc, math_approx_mode, dst_full_sync_en",
    [
        (16, 1, 16, 64, 115, 115, ttnn.bfloat8_b, ttnn.bfloat8_b, (4, 4), (1, 1), (0, 0), 16, 2 * 32, 1, True, ttnn.TILE_LAYOUT, ttnn.MathFidelity.LoFi, False, True, True, False,),         # works
        (16, 1, 16, 64, 115, 115, ttnn.bfloat8_b, ttnn.bfloat8_b, (4, 4), (1, 1), (0, 0), 16, 49 * 32, 1, True, ttnn.TILE_LAYOUT, ttnn.MathFidelity.LoFi, False, True, True, False,),         # does not work
        (16, 1, 16, 64, 115, 115, ttnn.bfloat8_b, ttnn.bfloat8_b, (4, 4), (1, 1), (0, 0), 16, 7 * 32, 1, True, ttnn.TILE_LAYOUT, ttnn.MathFidelity.LoFi, False, True, True, False,),         # does not work
        (16, 1, 16, 64, 115, 115, ttnn.bfloat8_b, ttnn.bfloat8_b, (4, 4), (1, 1), (0, 0), 16, 14 * 32, 1, True, ttnn.TILE_LAYOUT, ttnn.MathFidelity.LoFi, False, True, True, False,),         # does not work
        #(16, 1, 16, 64, 115, 115, ttnn.bfloat8_b, ttnn.bfloat8_b, (4, 4), (1, 1), (0, 0), 16, 1 * 32, 1, True, ttnn.TILE_LAYOUT, ttnn.MathFidelity.LoFi, False, True, True, False,),         # crash
    ],
)
@pytest.mark.parametrize("device_params", [{"l1_small_size": 16384}], indirect=True)
def test_resnet50_first_conv(
    device,
    use_program_cache,
    batch,
    groups,
    input_channels,
    output_channels,
    input_height,
    input_width,
    weights_dtype,
    activations_dtype,
    kernel,
    stride,
    padding,
    input_channels_alignment,
    act_block_h_override,
    act_block_w_div,
    deallocate_activation,
    output_layout,
    math_fidelity,
    fp32_accum,
    packer_l1_acc,
    math_approx_mode,
    dst_full_sync_en,
):
    torch.manual_seed(11234)

    conv_input_shape = [batch, input_channels, input_height, input_width]
    conv_weight_shape = [output_channels,  input_channels // groups, kernel[0], kernel[1]]
    conv_bias_shape = [1, 1, 1, output_channels]

    torch_input_tensor_nchw = torch.randn(conv_input_shape, dtype=torch.bfloat16).float()

    torch_weight_tensor = torch.randn(conv_weight_shape, dtype=torch.bfloat16).float()
    torch_bias_tensor = torch.randn(conv_bias_shape, dtype=torch.bfloat16).float()

    torch_out_golden_tensor = torch.nn.functional.conv2d(
        torch_input_tensor_nchw,
        torch_weight_tensor,
        bias=torch_bias_tensor.reshape(-1),
        stride=stride,
        padding=padding,
        dilation=(1, 1),
        groups=groups,
    )
    torch_out_golden_tensor = torch.nn.functional.relu(torch_out_golden_tensor)

    tt_weight_tensor = ttnn.from_torch(
        torch_weight_tensor, weights_dtype if weights_dtype != ttnn.bfloat8_b else ttnn.float32
    )
    tt_bias_tensor = ttnn.from_torch(
            torch_bias_tensor, weights_dtype if weights_dtype != ttnn.bfloat8_b else ttnn.float32
    )

    torch_input_tensor = torch.permute(torch_input_tensor_nchw, (0, 2, 3, 1))

    tt_input_tensor = ttnn.from_torch(torch_input_tensor, ttnn.bfloat16, mesh_mapper=None)
    tt_input_tensor = tt_input_tensor.reshape(
        1,
        1,
        tt_input_tensor.shape[0] * tt_input_tensor.shape[1] * tt_input_tensor.shape[2],
        tt_input_tensor.shape[3],
    )

    conv_config = ttnn.Conv2dConfig(
        dtype=activations_dtype,
        weights_dtype=weights_dtype,
        shard_layout=ttnn.TensorMemoryLayout.HEIGHT_SHARDED,
        input_channels_alignment=input_channels_alignment,
        activation="relu",
        deallocate_activation=deallocate_activation,
        enable_act_double_buffer=True,
        enable_split_reader=True,
        enable_subblock_padding=False,
        output_layout=output_layout,
        act_block_h_override=act_block_h_override,
        act_block_w_div=act_block_w_div,
        override_sharding_config = False,
        transpose_shards = False,
    )
    compute_config = ttnn.init_device_compute_kernel_config(
        device.arch(),
        math_fidelity=math_fidelity,
        fp32_dest_acc_en=fp32_accum,
        packer_l1_acc=packer_l1_acc,
        math_approx_mode=math_approx_mode,
        dst_full_sync_en=dst_full_sync_en,
    )

    tt_output_tensor, [out_height, out_width], [weight, bias] = ttnn.conv2d(
        input_tensor = tt_input_tensor,
        weight_tensor = tt_weight_tensor,
        bias_tensor = tt_bias_tensor,
        device = device,
        in_channels = input_channels,
        out_channels = output_channels,
        batch_size = batch,
        input_height = input_height,
        input_width = input_width,
        kernel_size = kernel,
        stride = stride,
        padding = padding,
        dilation=(1, 1),
        groups = groups,
        conv_config = conv_config,
        compute_config = compute_config,
        conv_op_cache = {},
        return_output_dim = True,
        return_weights_and_bias = True,
    )

    tt_output_tensor = ttnn.from_device(tt_output_tensor)
    torch_output_tensor = ttnn.to_torch(tt_output_tensor, mesh_composer=None)

    # NHWC to NCHW
    torch_output_tensor = torch_output_tensor.reshape(
        batch, out_height, out_width, torch_output_tensor.shape[-1]
    )
    torch_output_tensor = torch_output_tensor[:, :, :, :output_channels]
    torch_output_tensor = torch.permute(torch_output_tensor, (0, 3, 1, 2))

    target_pcc = 0.9997
    passing, pcc_msg = check_with_pcc_without_tensor_printout(torch_output_tensor, torch_out_golden_tensor, pcc=target_pcc)
    logger.info(f"PCC = {pcc_msg}. Threshold = {target_pcc}")
    assert passing, pcc_msg
```
